src/test/java/com/google/collegeplanner/DatastoreTest.java

```diff
@@ -0,0 +1,180 @@
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.collegeplanner;
+
+import static org.mockito.AdditionalMatchers.not;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.tools.development.testing.LocalDatastoreServiceTestConfig;
+import com.google.appengine.tools.development.testing.LocalServiceTestHelper;
+import com.google.collegeplanner.servlets.ApiUtil;
+import com.google.collegeplanner.servlets.DatastoreServlet;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.io.StringWriter;
+import java.net.URI;
+import java.net.URISyntaxException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import org.apache.http.client.utils.URIBuilder;
+import org.json.simple.JSONArray;
+import org.json.simple.JSONObject;
+import org.json.simple.parser.JSONParser;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import org.skyscreamer.jsonassert.JSONAssert;
+import org.skyscreamer.jsonassert.JSONCompareMode;
+
+/** Tests DatastoreServlet */
+@RunWith(JUnit4.class)
+public final class DatastoreTest {
+  HttpServletResponse response;
+  StringWriter stringWriter;
+  PrintWriter writer;
+  JSONParser parser;
+  DatastoreService datastore;
+
+  private final LocalServiceTestHelper helper =
+      new LocalServiceTestHelper(new LocalDatastoreServiceTestConfig());
+
+  @Before
+  public void before() throws Exception {
+    datastore = DatastoreServiceFactory.getDatastoreService();
+    response = mock(HttpServletResponse.class);
+    stringWriter = new StringWriter();
+    writer = new PrintWriter(stringWriter);
+    parser = new JSONParser();
+    when(response.getWriter()).thenReturn(writer);
+    helper.setUp();
+  }
+
+  @After
+  public void after() {
+    helper.tearDown();
+  }
+
+  /* Run this test twice to prove we're not leaking any state across tests. */
+  @Test
+  public void testDatastoreInsertion1() {
```

Comment:
curious - was this approach recommended somewhere? i'm not sure this is necessary.

Reply:
It was an example they gave in the docs but I guess it's not necessary

Reply:
I think just having any two tests below should provide the same functional coverage of this case.